third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -301,7 +301,17 @@ impl<'a> Transformer<'a> {
                         let new_srcs = self.copy_args_if_needed(code_offset, id, srcs);
                         self.check_and_emit_bytecode(code_offset, Call(id, dests, op, new_srcs, ai))
                     },
-                    _ => self.check_and_emit_bytecode(code_offset, bc.clone()),
+                    _ => {
+                        if op.can_take_non_ref_primitive_as_args() {
+                            let new_srcs = self.copy_args_if_needed(code_offset, id, srcs);
```

Comment:
Would it be possible to rewrite this so that we don't have code duplications (the bodies are the same for `Function(..)` and this case)?